libraries/AP_HAL/examples/Printf/Printf.cpp

```diff
@@ -1,19 +1,41 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+/*
+    Printf.cpp: We demonstrate the use of the printf() and snprintf() functions
+*/
 #include <AP_Common/AP_Common.h>
 #include <AP_HAL/AP_HAL.h>
 
-void setup();
-void loop();
+void setup();    //declaration of the setup() function
+void loop();     //declaration of the loop() function
 
-const AP_HAL::HAL& hal = AP_HAL::get_HAL();
+const AP_HAL::HAL& hal = AP_HAL::get_HAL();    //create a reference to AP_HAL::HAL object to get access to hardware specific functions. For more info see <https://ardupilot.org/dev/docs/learning-ardupilot-the-example-sketches.html/>
 
+// the setup function runs once when the board powers up
 void setup(void) {
-    hal.console->printf("Starting Printf test\n");
+    hal.console->printf("Starting Printf test\n");    //print a starting message
 }
 
+//create a struct array float_tests
```

Comment:
comment need a space after //

Reply:
Thank you, Sir, for the suggestion. I will work on the changes